samples/SimpleDataSource/HelloWorldCustomDataProcessor.cs

```diff
@@ -0,0 +1,159 @@
+    //
+    // Data processor is responsible for instantiating the proper tables based on what the user has decided to enable.
+    //
+
+    //
+    // Derive the CustomDataProcessorBase abstract class.
+    //
+
+    public sealed class HelloWorldCustomDataProcessor
+        : CustomDataProcessorBase
+    {
+        private readonly string[] filePaths;
+        private IReadOnlyDictionary<string, IReadOnlyList<Tuple<Timestamp, string>>> fileContent;
+        private DataSourceInfo dataSourceInfo;
+
+        public HelloWorldCustomDataProcessor(
+           string[] filePaths,
+           ProcessorOptions options,
+           IApplicationEnvironment applicationEnvironment,
+           IProcessorEnvironment processorEnvironment,
+           IReadOnlyDictionary<TableDescriptor, Action<ITableBuilder, IDataExtensionRetrieval>> allTablesMapping,
+           IEnumerable<TableDescriptor> metadataTables)
+            : base(options, applicationEnvironment, processorEnvironment, allTablesMapping, metadataTables)
+        {
+            //
+            // Assign the files array to a readonly backing field.
+            //
+
+            this.filePaths = filePaths;
+        }
+
+        public override DataSourceInfo GetDataSourceInfo()
+        {
+            //
+            // The DataSourceInfo is used to tell the SDK the time range of the data(if applicable) and any other relevant data for rendering / synchronizing.
+            // In our case, we have nothing, so just return new DataSourceInfo(0, long.MaxValue) to tell the SDK that it encompasses all time.
+            //
+
+            return this.dataSourceInfo;
+            
+        }
+
+        protected override Task ProcessAsyncCore(
+           IProgress<int> progress,
+           CancellationToken cancellationToken)
+        {
+            //
+            // This is where you add your own logic to process the data into a format for your tables.
+            //
+            // In this sample, our tables will operate on a collection of lines in the file
+            // so we read all of the lines of the file and store them into a backing dictionary field.
+            //
+            // ProcessAsync is also where you would determine the information necessary for GetDataSourceInfo().
+            // In this sample, we take down the start and stop timestamps from the files
+            //
+            // Note: if you must do processing based on which tables are enabled, you would check the EnabledTables property (provided in the base class) on your class to see what you should do.
+            // For example, XPerf looks at what tables are enabled in order to turn on additional event processors to avoid processing everything if it doesn't have to.
+            //
+            // In this sample we create a Time Stamp column that can be used for graphing. For more information on graphing, go to "Graphing" on our Wiki. Link can be found in README.md
```

Comment:
this  method doesn't actually create any columns, so maybe connect this statement to the startTime, endTime, and firstEvent locals below the comment

Reply:
yeah, this commend should be deleted